archunit/src/main/java/com/tngtech/archunit/library/Architectures.java

```diff
@@ -289,4 +291,114 @@ public String toString() {
             }
         }
     }
+
+    @PublicAPI(usage = ACCESS)
+    public static OnionArchitecture onionArchitecture() {
+        return new OnionArchitecture();
+    }
+
+    public static final class OnionArchitecture implements ArchRule {
+        private static final String DOMAIN_MODEL_LAYER = "domain model";
+        private static final String DOMAIN_SERVICE_LAYER = "domain service";
+        private static final String APPLICATION_LAYER = "application";
+        private static final String ADAPTER_LAYER = "adapter";
+
+        private final Set<String> domainModelPackageIdentifiers = new LinkedHashSet<>();
```

Comment:
As they're only used as varargs, I think you could also store them in an array. I don't think using a `HashSet` of any kind provides a benefit.

Reply:
done :)